attachment_azure/models/ir_attachment.py

```diff
@@ -0,0 +1,165 @@
+            msg = _(
+                "If you want to read from the Azure container, you must provide the "
+                "following environment variables:\n"
+                "* AZURE_STORAGE_CONNECTION_STRING\n"
+                "or\n"
+                "* AZURE_STORAGE_ACCOUNT_NAME\n"
+                "* AZURE_STORAGE_ACCOUNT_URL\n"
+                "* AZURE_STORAGE_ACCOUNT_KEY\n"
+            )
+            raise exceptions.UserError(msg)
+        blob_service_client = None
+        if connect_str:
+            try:
+                blob_service_client = BlobServiceClient.from_connection_string(
+                    connect_str
+                )
+            except exceptions.UserError as error:
+                _logger.exception(
+                    "Error during the connection to Azure container using the "
+                    "connection string."
+                )
+                raise exceptions.UserError(str(error))
+        else:
+            try:
+                sas_token = generate_account_sas(
+                    account_name=account_name,
+                    account_key=account_key,
+                    resource_types=ResourceTypes(service=True),
+                    permission=AccountSasPermissions(read=True),
+                    expiry=datetime.utcnow() + timedelta(hours=1),
+                )
+                blob_service_client = BlobServiceClient(
+                    account_url=account_url,
+                    credential=sas_token,
+                )
+            except exceptions.UserError as error:
+                _logger.exception(
+                    "Error during the connection to Azure container using the Shared "
+                    "Access Signature (SAS)"
+                )
+                raise exceptions.UserError(str(error))
+        return blob_service_client
+
+    @api.model
+    def _get_azure_container(self):
+        running_env = os.environ.get("RUNNING_ENV", "dev")
+        container_name = str.lower(running_env + "-" + self.env.cr.dbname)
+        blob_service_client = self._get_blob_service_client()
+        container_client = blob_service_client.get_container_client(container_name)
+        try:
+            # Create the container
+            container_client.create_container()
+        except ResourceExistsError:
+            pass
+        except exceptions.UserError as error:
+            _logger.exception("Error during the creation of the Azure container")
+            raise exceptions.UserError(str(error))
+        return container_client
+
+    @api.model
+    def _store_file_read(self, fname, bin_size=False):
+        if fname.startswith("azure://"):
+            container_client = self._get_azure_container()
+            key = fname.replace("azure://", "", 1).lower()
+            try:
+                blob_client = container_client.get_blob_client(key)
+                read = blob_client.download_blob().readall()
+            except exceptions.UserError:
```

Comment:
This catch seems suspicious, `container_client` and `blob_client` are azure lib objects, I strongly doubt it raises Odoo like exceptions.